src/hooks/useMutex.js

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+function useMutex() {
```

Comment:
A mutex is used to keep navigation from unmounting element and slide undergoing animations causing a state update exception. React spring does not have the concept of time (as it is a spring-based animation library) so the animations are run-to-completion.